### PR TITLE
Move roles source of truth in the configuration

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
The source of truth for the roles is now the configuration
This is mainly done so we can push roles via a configuration file for multi-cluster management

# Impacts
Roles CRUD

# Delete branch after merge
YES

# NEWS file entries
## Enhancements
* Roles are now stored in a configuration file for easier backup and management

# UPGRADE file entries

The source of truth for roles is now in a configuration file. In order to pull the existing roles from your database into the configuration file, execute the following command:

```
# /usr/local/pf/addons/upgrade/to-7.0-roles-conf.pl
```

### CODE REVIEW

**Comment by dwlfrth** [comment]
Rebase me plz

**Comment by julsemaan** [comment]
rhibazaidh welcome

File: html/pfappserver/lib/pfappserver/Model/Config/Roles.pm [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+=head1 NAME
+
+pfappserver::Model::Config::Roles
+
+=cut
+
+=head1 DESCRIPTION
+
+pfappserver::Model::Config::Roles
```
Comment (dwlfrth) [inline html/pfappserver/lib/pfappserver/Model/Config/Roles.pm]: Cute description ;)
bhahaha

File: lib/pf/ConfigStore/Roles.pm [hunk 2]
```diff
@@ -0,0 +1,73 @@
+package pf::ConfigStore::Roles;
+=head1 NAME
+
+pf::ConfigStore::Roles
+Store Roles configuration
```
Comment (dwlfrth) [inline lib/pf/ConfigStore/Roles.pm]: Name , Description

File: lib/pf/ConfigStore/Roles.pm [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+=cut
+
+=head1 DESCRIPTION
```
Comment (dwlfrth) [inline lib/pf/ConfigStore/Roles.pm]: Description, name, all mixed up !

**Review by dwlfrth — COMMENTED** [review]
Appart from the small comments, if it's working I don't see a reason why not merging it
👯

**Comment by julsemaan** [comment]
@dwlfrth, unmixed the mixed perldoc that was a bit too mixed up

**Pull request merged**